$OpenBSD: patch-gio_tests_file_c,v 1.1 2012/10/12 20:52:04 ajacoutot Exp $

XXX why isn't this working with kqueue?
push upstream when the kqueue patch is committed there

--- gio/tests/file.c.orig	Mon Sep  3 16:24:05 2012
+++ gio/tests/file.c	Tue Oct  9 23:47:22 2012
@@ -424,11 +424,12 @@ test_create_delete (gconstpointer d)
   data->monitor = g_file_monitor_file (data->file, 0, NULL, &error);
   g_assert_no_error (error);
 
-  /* This test doesn't work with GPollFileMonitor, because it assumes
-   * that the monitor will notice a create immediately followed by a
-   * delete, rather than coalescing them into nothing.
+  /* This test doesn't work with GPollFileMonitor nor GKqueueDirectoryMonitor,
+   * because it assumes that the monitor will notice a create immediately
+   * followed by a delete, rather than coalescing them into nothing.
    */
-  if (!strcmp (G_OBJECT_TYPE_NAME (data->monitor), "GPollFileMonitor"))
+  if ((!strcmp (G_OBJECT_TYPE_NAME (data->monitor), "GPollFileMonitor")) ||
+      (!strcmp (G_OBJECT_TYPE_NAME (data->monitor), "GKqueueDirectoryMonitor")))
     {
       g_print ("skipping test for this GFileMonitor implementation");
       goto skip;
